app/src/main/java/com/android/sample/model/workout/WorkoutRepositoryFirestore.kt

```diff
@@ -133,7 +142,20 @@ open class WorkoutRepositoryFirestore<T : Workout>(
    * @param onSuccess Callback triggered with the list of workout documents on success.
    * @param onFailure Callback triggered with an exception on failure.
    */
-  override fun getDocuments(onSuccess: (List<T>) -> Unit, onFailure: (Exception) -> Unit) {
+  override suspend fun getDocuments(onSuccess: (List<T>) -> Unit, onFailure: (Exception) -> Unit) {
```

Comment:
Here I see that you always take what is in the cache, if the cache has some information. Indeed, you also correctly update the cache when you upload something to firestore, but in the case that the workout is changed in the backend, there will be discrepancies between the data the user gets and the data on the server. Maybe it would be good to have some fetches from time to time to ensure freshness. This would have been good to have for a feature where a workout is shared between the users, but I do not think we have time to implement it, so this issue is not top priority.

Reply:
Great idea but you're right since we don't have that feature yet it wouldn't change much because I coded this thinking that the only way to modify the backend is through the own user's UI which then modifies the cache